opm/autodiff/RedistributeDataHandles.hpp

```diff
@@ -40,12 +40,84 @@ inline void distributeGridAndData( Grid& ,
                                    BlackoilPropsAdFromDeck& ,
                                    DerivedGeology&,
                                    std::shared_ptr<BlackoilPropsAdFromDeck::MaterialLawManager>&,
+                                   std::vector<double>&,
                                    boost::any& ,
                                    const bool )
 {
 }
 
 #if HAVE_DUNE_CORNERPOINT && HAVE_MPI
+/// \brief a data handle to distribute the threshold pressures
+class ThresholdPressureDataHandle
```

Comment:
This class seems to me to work on a cell level, and then for each cell communicating the threshold pressures for each adjacent face. That would communicate almost twice as many values as necessary. Could that be reduced by communicating on faces instead?